### ISSUE TITLE
0.17 beta 10/12, jspm build npm:tweetnacl@0.14.1 --node hangs

### ISSUE THREAD

**Issue by AdamBurgess** [jspm/jspm-cli#1599]
to reproduce:
`npm init -y && npm install jspm@beta && jspm init -y && jspm install npm:tweetnacl@0.14.1 && jspm build npm:tweetnacl@0.14.1 --node`
this has only broke recently; today I had locked my npm dependencies to jspm-cli@6cfc89c and systemjs/builder@08f69b90a11 where `jspm build --node` still worked

running node 5.3.0, windows 10 x64.

--
originally I thought it was request.
`jspm build npm:request@2.69.0` works, `jspm build npm:request@2.69.0 --node` hangs indefinitely, using 100% CPU.

note: it's likely a dependency of request that isn't working, not request itself
edit: request@2.69.0 relies on npm:http-signature@1.1.1, which relies on npm:sshpk@1.7.4, which implicitly relies on npm:tweetnacl@0.14.1, which hangs! `npm:tweetnacl@0.14.1` is the baddie

**Comment by guybedford** [comment]
Thanks, fixed in https://github.com/systemjs/builder/commit/1c0053fcd27a1603751f7735c02c1adc8fdf49e8.

**Comment by AdamBurgess** [comment]
With your patch I now get:
```
O:\jspm\15>jspm build npm:tweetnacl@0.14.1 --node
     Creating the single-file build for npm:tweetnacl@0.14.1...

err  TypeError: No resolution found at all for condition npm:tweetnacl@0.14.1/#:crypto.
    at O:\jspm\15\node_modules\systemjs-builder\lib\trace.js:668:15
    at Array.forEach (native)
    at O:\jspm\15\node_modules\systemjs-builder\lib\trace.js:661:6
```

**Comment by guybedford** [comment]
Thanks, also fixed in https://github.com/systemjs/builder/commit/f83f52e9078c91a7dbee9a55a744a3915fc4e957.

**Comment by AdamBurgess** [comment]
awesome, seems to work.

Although I am curious that it doesn't require nodelibs-crypto, as nacl-fast.js does (I think?)
```js
} else if (typeof require !== 'undefined') {
  // Node.js.
  crypto = require('crypto');
  if (crypto && crypto.randomBytes) {
    nacl.setPRNG(function(x, n) {
      var i, v = crypto.randomBytes(n);
      for (i = 0; i < n; i++) x[i] = v[i];
      cleanup(v);
    });
  }
}
```

**Comment by AdamBurgess** [comment]
On a second look, the require is changed to `@node/require` in the build.js, so I guess systemjs is being smart and using node's standard require :+1:

**Comment by guybedford** [comment]
Released in 0.17.0-beta.11.

**Comment by guybedford** [comment]
And yes exactly - the conditional system will defer to Node core crypto when running in Node, which it can inline when doing a Node build. Most of the dependency graph analysis code actually deals with handling all these conditional features!

**Comment by AdamBurgess** [comment]
@guybedford, this bug has resurfaced in beta 12 -- exact same repro steps.

**Comment by guybedford** [comment]
Thanks, yes this fix was actually itself a regression, will look into it again.

**Comment by guybedford** [comment]
I've fixed this properly in https://github.com/systemjs/builder/commit/8bac50b3b85e3365b921d0a37dd99c5125064423.

**Comment by guybedford** [comment]
The reason this was switching back and fourth was actually a good one - fallbacks normalize differently to map targets, which needed to be reflected here... hopefully cracked in https://github.com/systemjs/builder/commit/6f25cda554ff40e3f1ef9d45bc1c11df83dae17d.

**Comment by guybedford** [comment]
Released in 0.17.0-beta.13.
